File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.go
func canBeEqual(targetArray []int, currentArray []int) bool {
    elementCounts := make([]int, 1001)
    uniqueCount := 0
    
    for i := 0; i < len(targetArray); i++ {
        if elementCounts[targetArray[i]] == 0 {
            uniqueCount++
        }
        elementCounts[targetArray[i]]++
        
        if elementCounts[currentArray[i]] == 1 {
            uniqueCount--
        }
        elementCounts[currentArray[i]]--
    }
    
    return uniqueCount == 0
}

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.js
/**
 * @param {number[]} targetArray
 * @param {number[]} currentArray
 * @return {boolean}
 */
var canBeEqual = function(targetArray, currentArray) {
    const elementCounts = new Array(1001).fill(0);
    let uniqueCount = 0;
    
    for (let i = 0; i < targetArray.length; i++) {
        if (elementCounts[targetArray[i]]++ === 0) uniqueCount++;
        if (elementCounts[currentArray[i]]-- === 1) uniqueCount--;
    }
    
    return uniqueCount === 0;
};